doc/user/content/sql/begin.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "BEGIN"
+menu:
+  main:
+    parent: "sql"
+---
+
+`BEGIN` starts a transaction block.
+
+## Syntax
+
+{{< diagram "begin.svg" >}}
+
+Supported `transaction_mode` option values:
+
+Value | Description
+------|----------
+`ISOLATION LEVEL SERIALIZABLE` | A no-op because transactions are always serializable.
+
+## Details
+
+`BEGIN` starts a transaction block.
+All statements in a transaction block will be executed in a single transaction until an explicit `COMMIT` or `ROLLBACK` is given.
+
+Transactions in Materialize do not support interleaving arbitrary kinds of statements, but instead are either **read only** or **write only**, determined by the first statement after the `BEGIN`.
+
+A **read-only** transaction starts with a [`SELECT`](/sql/select) statement and allows only `SELECT` statements.
+Because Materialize does not know which objects (sources, tables, or views) will be queried during the transaction, the objects in the first `SELECT` and any other object in the same schemas are assumed to be possible query targets.
+Other queries can only reference these objects.
+During the first query, a timestamp is chosen that is valid for all of those objects.
+This timestamp will be used for all other queries.
+The transaction will additionally hold back normal compaction of the objects, potentially increasing memory usage for very long running transactions.
+If `AS OF` is specified in any query, it is exempt from the above restrictions.
```

Comment:
Does this include exemption from _all_ of the restrictions, including the potential limiting of references?

Reply:
Yes, I've added clarifying text.